pkg/controller/daemon/daemon_controller.go

```diff
@@ -801,6 +801,70 @@ func (dsc *DaemonSetsController) resolveControllerRef(namespace string, controll
 	return ds
 }
 
+// syncPodOnNode syncs the DaemonSet Pods on the node, the return values are:
+//   - nodesNeedingDaemonPods: the pods need to start on the node
+//   - podsToDelete: the Pods need to be deleted on the node
+//   - failedPodsObserved: the number of failed pods on node
+//   - err: unexpected error
+func (dsc *DaemonSetsController) syncPodOnNode(
```

Comment:
nit: Need to rename this function. "Sync" means reconciles the current state to make it match the given desired state (e.g. `syncNodes` in this file syncs by creating/deleting pods on nodes). 

What this function does is calculating the diff. Suggest naming it something like "podsShouldOnNode". The comment can be something like "podsShouldOnNode figures out the DaemonSet pods to be created and deleted on the given node".

Reply:
done :)